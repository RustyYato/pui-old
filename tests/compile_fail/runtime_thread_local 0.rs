pui::make_counter_tl! {
    type Foo = u8;
}

fn main() {
    fn assert_send_sync<T: Send + Sync>(_: T) {}
    assert_send_sync(Foo::new());
}
