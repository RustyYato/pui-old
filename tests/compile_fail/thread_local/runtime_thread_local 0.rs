pui::make_global_id_alloc_tl! {
    type Foo = u8;
}

fn main() {
    fn assert_send_sync<T: Send + Sync>(_: T) {}
    assert_send_sync(Foo::new());
}
