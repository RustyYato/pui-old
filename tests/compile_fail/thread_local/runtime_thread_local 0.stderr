error: no rules expected the token `=`
 --> $DIR/runtime_thread_local 0.rs:2:14
  |
2 |     type Foo = u8;
  |              ^ no rules expected this token in macro call

error[E0433]: failed to resolve: use of undeclared type `Foo`
 --> $DIR/runtime_thread_local 0.rs:7:22
  |
7 |     assert_send_sync(Foo::new());
  |                      ^^^ use of undeclared type `Foo`
